bag_verifier/src/main/scala/uk/ac/wellcome/platform/archive/bagverifier/services/s3/S3BagVerifier.scala

```diff
@@ -71,16 +62,62 @@ class S3StandaloneBagVerifier(val primaryBucket: String)(
   ): Either[BagVerifierError, Unit] = Right(())
 }
 
-class S3ReplicatedBagVerifier(val primaryBucket: String)(
-  implicit val s3Client: AmazonS3
+class S3ReplicatedBagVerifier(
+  val primaryBucket: String,
+  val bagReader: BagReader[S3ObjectLocation, S3ObjectLocationPrefix],
+  val listing: Listing[S3ObjectLocationPrefix, S3ObjectLocation],
+  val resolvable: Resolvable[S3ObjectLocation],
+  val fixityListChecker: FixityListChecker[
+    S3ObjectLocation,
+    S3ObjectLocationPrefix,
+    Bag
+  ],
+  val srcStreamStore: StreamStore[S3ObjectLocation],
+  val replicaStreamStore: StreamStore[S3ObjectLocation]
 ) extends ReplicatedBagVerifier[
       S3ObjectLocation,
       S3ObjectLocationPrefix
     ]
     with S3BagVerifier[
       ReplicatedBagVerifyContext[S3ObjectLocationPrefix]
-    ] {
+    ]
 
-  override val replicaStreamStore: StreamStore[S3ObjectLocation] =
-    new S3StreamStore()
+object S3BagVerifier {
```

Comment:
Minor: I would have made objects for each of S3StandaloneBagVerifier and S3ReplicatedBagVerifier, each with a single apply() method, but I'm not sure it makes a difference.